routers/repo/pull.go

```diff
@@ -679,7 +679,12 @@ func MergePullRequest(ctx *context.Context, form auth.MergePullRequestForm) {
 		return
 	}
 	if issue.IsClosed {
-		ctx.NotFound("MergePullRequest", nil)
+		if issue.IsPull {
+			ctx.Flash.Error(ctx.Tr("repo.pulls.is_closed"))
+		} else {
+			ctx.Flash.Error(ctx.Tr("repo.issues.closed_title"))
+		}
+		ctx.Redirect(ctx.Repo.RepoLink + "/pulls/" + com.ToStr(issue.Index))
```

Comment:
Why not set issue/pull specific redirect **in** if statement?

Reply:
I know it works without - but if you read throu the code this is weared

Reply:
Done